pkg/operator/controllers/checker/auth.go

```diff
@@ -0,0 +1,72 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	jwt "github.com/form3tech-oss/jwt-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/validate"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+type credentials struct {
+	clientID       []byte
+	clientSecret   []byte
+	tenantID       []byte
+	subscriptionID []byte
+}
+
+func newAuthorizer(token *adal.ServicePrincipalToken) (refreshable.Authorizer, error) {
+	p := &jwt.Parser{}
+	c := &validate.AzureClaim{}
+	_, _, err := p.ParseUnverified(token.OAuthToken(), c)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, role := range c.Roles {
+		if role == "Application.ReadWrite.OwnedBy" {
+			return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalCredentials, "properties.servicePrincipalProfile", "The provided service principal must not have the Application.ReadWrite.OwnedBy permission.")
+		}
+	}
+
+	return refreshable.NewAuthorizer(token), nil
+}
+
+func azCredentials(ctx context.Context, kubernetescli kubernetes.Interface) (*credentials, error) {
+	var creds credentials
+	mysec, err := kubernetescli.CoreV1().Secrets(azureCredentialSecretNamespace).Get(ctx, azureCredentialSecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if _, ok := mysec.Data["azure_client_id"]; !ok {
+		return nil, errors.New("azure_client_id does not exist")
+	}
+	creds.clientID = mysec.Data["azure_client_id"]
```

Comment:
nit: you already getting value but discarding it by doing `_, ok`. We could use it for assignment . Sometimes (not saying this is the time) this looks more readable.